sys/net/network_layer/ng_ipv6/netif/ng_ipv6_netif.c

```diff
@@ -441,6 +443,65 @@ static ng_ipv6_addr_t *_match_prefix(kernel_pid_t pid, const ng_ipv6_addr_t *add
     return NULL;
 }
 
+/**
+ * @brief selects potential source address candidates
+ * see http://tools.ietf.org/html/rfc6724#section-4
+ * @param[in]  iface            the interface used for sending
+ * @param[in]  dest             the destination address
+ * @param[out] candidate_set    a bitfield representing all addresses
+ *                              configured to @p iface, potential candidates
+ *                              will be marked as 1
+ *
+ * @return false if no candidates were found
+ * @return true otherwise
+ *
+ * @pre the interface entry and its set of addresses must not be changed during
+ *      runtime of this function
+ */
+static bool _create_candidate_set(ng_ipv6_netif_t *iface,
+                                  const ng_ipv6_addr_t *dest,
+                                  uint8_t *candidate_set)
+{
+    for (int i = 0; i < NG_IPV6_NETIF_ADDR_NUMOF; i++) {
+        ng_ipv6_addr_t *iter = &(iface->addrs[i].addr);
```

Comment:
you might need to utilize the mutex of `ng_ipv6_netif_t`, before accessing any further members of this struct